src/app/shared/directives/postal-code-validator.directive.spec.ts

```diff
@@ -1,8 +1,36 @@
-import { PostalCodeValidatorDirective } from './postal-code-validator.directive';
+import { FormControl } from '@angular/forms';
+import { PostalCodeValidator, PostalCodeValidatorDirective } from './postal-code-validator.directive';
 
 describe('PostalCodeValidatorDirective', () => {
-  it('should create an instance', () => {
+  it('should create an instance of the directive', () => {
     const directive = new PostalCodeValidatorDirective();
     expect(directive).toBeTruthy();
   });
+
+  describe('PostalCodeValidator', () => {
```

Comment:
Because we say postal code here and not ZIP, you should know that there are many other ways a postal code can be valid. It might be a fun exercise to support more countries or you may make the decision to only validate US postal codes. But what about a +4?

https://en.wikipedia.org/wiki/List_of_postal_codes